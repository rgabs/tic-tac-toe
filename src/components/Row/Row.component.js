import Cell from '../Cell/Cell.component';
import PropTypes from 'prop-types';
import React from 'react';
import noop from 'lodash/noop';

const Row = ({cellData, cellclickHandler, rowID}) =>
  <div className='row'>
    <Cell value={cellData[0]} cellclickHandler={cellclickHandler(rowID, 0)}/>
    <Cell value={cellData[1]} cellclickHandler={cellclickHandler(rowID, 1)}/>
    <Cell value={cellData[2]} cellclickHandler={cellclickHandler(rowID, 2)}/>
  </div>;

Row.propTypes = {
  'cellclickHandler': PropTypes.func.isRequired,
  'cellData': PropTypes.array.isRequired,
  'rowID' : PropTypes.string.isRequired
};
Row.defaultProps = {
  'cellclickHandler': noop,
  'cellData': '',
  'rowID': 'XX'
};
export default Row;
