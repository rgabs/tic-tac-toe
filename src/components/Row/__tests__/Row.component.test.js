import React from 'react';
import Row from '../Row.component';
import noop from 'lodash/noop';
import renderer from 'react-test-renderer';
test('Row: snapshot test', () => {
  const cData = ['X', 'O', 'X'];
  const snapshot = renderer.create(<Row cellData={cData} rowID={'1'}
    cellclickHandler={noop}/>).toJSON();
  expect(snapshot).toMatchSnapshot();
});
