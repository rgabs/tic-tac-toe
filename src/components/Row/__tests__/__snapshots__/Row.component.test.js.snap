// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row: snapshot test 1`] = `
<div
  className="row"
>
  <div
    className="cell disabled-cell"
    onClick={[Function]}
  >
    X
  </div>
  <div
    className="cell disabled-cell"
    onClick={[Function]}
  >
    O
  </div>
  <div
    className="cell disabled-cell"
    onClick={[Function]}
  >
    X
  </div>
</div>
`;
